### PR TITLE
Allow compiling with -Wfloat-conversion

### PR DESCRIPTION
**Pull request by jendrikseipp**

### CODE REVIEW

File: parallel_hashmap/phmap.h [hunk 1]
```diff
@@ -1806,7 +1806,7 @@ class raw_hash_set
 
     size_t bucket_count() const { return capacity_; }
     float load_factor() const {
-        return capacity_ ? static_cast<double>(size()) / capacity_ : 0.0;
+        return capacity_ ? static_cast<float>(size()) / capacity_ : 0.0f;
```
Comment (greg7mdp) [inline parallel_hashmap/phmap.h:1809]: Would this work?

```suggestion
        return capacity_ ? static_cast<float>(static_cast<double>(size()) / capacity_) : 0.0f;
```

Comment (jendrikseipp) [reply parallel_hashmap/phmap.h:1809]: Yes, seems to do the trick.

Comment (greg7mdp) [reply parallel_hashmap/phmap.h:1809]: What compiler do you see this warning on that line? I don't see with with clang-17 or gcc-12.

Comment (jendrikseipp) [reply parallel_hashmap/phmap.h:1809]: I'm using GCC 11.4 with -Wfloat-conversion.

**Comment by greg7mdp** [comment]
Thanks for the fix @jendrikseipp

**Pull request merged**